Sources/NIOFileSystem/Internal/System Calls/Syscall.swift

```diff
@@ -31,8 +31,8 @@ import CNIOLinux
 @_spi(Testing)
 public enum Syscall: Sendable {
     @_spi(Testing)
-    public static func stat(path: FilePath) -> Result<CInterop.Stat, Errno> {
-        path.withPlatformString { platformPath in
+    public static func stat(path: NIOFilePath) -> Result<CInterop.Stat, Errno> {
```

Comment:
I don't think you really needed to change any of these. The purpose of `NIOFilePath` is to be at the API layer. Internally `FilePath` is an implementation detail so the simplest strategy would be to just drop down to `FilePath` as soon as possible.

Note also that these are `@_spi(Testing)` so not actually `public` API.